core/common/connector-core/src/main/java/org/eclipse/edc/connector/core/CoreDefaultServicesExtension.java

```diff
@@ -84,18 +91,27 @@ public EventExecutorServiceContainer eventExecutorServiceContainer() {
     }
 
     @Provider(isDefault = true)
-    public Vault vault() {
-        return new NoopVault();
+    public Vault vault(ServiceExtensionContext context) {
+        var seedSecrets = context.getSetting(VAULT_MEMORY_SECRETS_PROPERTY, null);
+        var vault = getVault(context);
+        if (seedSecrets != null) {
+            Stream.of(seedSecrets.split(SECRET_SEPARATOR))
+                    .filter(pair -> pair.contains(SECRET_KEY_VAULE_SEPARATOR))
+                    .map(kvp -> kvp.split(SECRET_KEY_VAULE_SEPARATOR, 2))
+                    .filter(kvp -> kvp.length >= 2)
+                    .forEach(pair -> vault.storeSecret(pair[0], pair[1]));
+        }
```

Comment:
I think we do, because if we don't have this, then there is no way to store secrets ahead of time, which may be required for (e2e-) testing.

Reply:
> I think we do, because if we don't have this, then there is no way to store secrets ahead of time, which may be required for (e2e-) testing.

unless it's needed I'd say that's not needed, and in that case I'd use a custom extension used only by the tests